# Hypothesis Testing

In [1]:
#importing the necessary libraires' 
import pandas as pd
import scipy as sc
import numpy as np 
from scipy import stats

In [2]:
# so here it is 1 sample and 1 tail test as we are doing it from one sample data and determinig the weekly cost predicted by the cost model is higher or not
t=(3050-4000)/(125/np.sqrt(25))
t

-38.0

In [3]:
p=stats.t.cdf(t,df=24) # as the sample mean is less than theoretical mean we are gonna go with stats.t.cdf
alpha=0.05
p

2.9607810808177907e-23

In [4]:
critical_value_one_tailed = stats.norm.ppf(1 - alpha)# standard normal Z distribution critical value
critical_value_one_tailed

1.6448536269514722

In [5]:
if p>alpha:
    print("Accept the Null hypothesis, μ ≤ 4000")
else:
    print("Reject the Null Hypothesis, μ > 4000")

Reject the Null Hypothesis, μ > 4000


## This conclusion suggests that further investigation or adjustments to the cost model may be necessary to align with the observed data from the franchise operations.




# Chi-Square Test

In [6]:
#importing required function from the scipy librabry
from scipy.stats import chi2_contingency


In [15]:
data=[[50,80,60,30,20],[70,100,90,50,50]]

In [14]:
chi,p,df,expected = chi2_contingency(data)
chi2_contingency(data)

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.,  60.,  32.,  28.],
       [ 72., 108.,  90.,  48.,  42.]]))

### the Chi-Square test will output dof=4, not 1, because the data has 2 categories for Smart home device type and 5 categories for Customer satisfaction level, resulting in (2−1)×(5−1)=4 degrees of freedom.










In [18]:
alpha = 0.05
print("P value is " +str(p))
if p>alpha:
    print("The type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level are independent of each other(Accept Null Hypothesis)")
else:
    print("They are related (Reject Null Hypothesis)")

P value is 0.22784371130697179
the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level are independent of each other(Accept Null Hypothesis)


In [22]:
from scipy.stats import chi2
dof=4
alpha=0.05
critical_value = chi2.ppf(1 - alpha, dof)
critical_value

9.487729036781154

### The Chi-Square test (χ² = 5.638227513227513 , p =0.22784371130697179 , dof = 4) indicates that we fail to reject  the null hypothesis at α = 0.05, suggesting that there is not a significant association between the type of smart home device purchased and customer satisfaction levels.

